File: cs/sdk/src/Proton.Drive.Sdk/Nodes/DegradedFolderNode.cs
using Proton.Sdk;

namespace Proton.Drive.Sdk.Nodes;

public sealed record DegradedFolderNode : DegradedNode
{
    public FolderNode ToNode(string substituteName)
    {
        return new FolderNode
        {
            Uid = Uid,
            ParentUid = ParentUid,
            Name = Name.TryGetValue(out var name) ? name : substituteName,
            NameAuthor = NameAuthor,
            CreationTime = CreationTime,
            TrashTime = TrashTime,
            Author = Author,
        };
    }
}

File: kt/sdk/src/main/kotlin/me/proton/drive/sdk/entity/DegradedFolderNode.kt
package me.proton.drive.sdk.entity

import java.time.Instant

data class DegradedFolderNode(
    override val uid: NodeUid,
    override val parentUid: ParentNodeUid?,
    override val treeEventScopeId: ScopeId,
    override val name: Result<String>,
    override val creationTime: Instant,
    override val trashTime: Instant?,
    override val nameAuthor: Result<Author>,
    override val author: Result<Author>,
    override val errors: List<DriveError>,
) : DegradedNode
